record-and-playback/presentation/scripts/process/presentation.rb

```diff
@@ -198,12 +202,25 @@
       FileUtils.cp_r("#{pres_dir}/thumbnails", "#{target_pres_dir}/thumbnails")
     end
 
-    BigBlueButton.logger.info("Generating closed captions")
-    ret = BigBlueButton.exec_ret('utils/gen_webvtt', '-i', raw_archive_dir, '-o', target_dir)
-    if ret != 0
-      raise "Generating closed caption files failed"
+    if File.exist?("#{captions_meeting_dir}/captions.json")
+      BigBlueButton.logger.info("Copying closed captions")
+
+      captions = JSON.load(File.new("#{captions_meeting_dir}/captions.json"))
+      captions_json = []
+      captions.each do |track|
+        caption = {}
+        caption[:localeName] = track['label']
+        caption[:locale] = track['lang']
+        captions_json << caption
+        FileUtils.cp("#{captions_meeting_dir}/caption_" + track['lang'] + ".vtt", target_dir)
+      end
+
+      File.open("#{target_dir}/captions.json", "w") do |f|
+        f.write(captions_json.to_json)
+      end
```

Comment:
The file `#{target_dir}/captions.json` must always be created, even if the meeting did not have any captions. If there are no captions it should contain an empty array.